### DISCUSSION TITLE
48. 데이터 캡슐화 과정에서 헤더와 트레일러가 모두 붙는 계층과 트레일러가 붙는 이유에 대해 설명해 주세요.

### DISCUSSION THREAD

**Discussion by linirini** [jewelry-box/cs-interview#50] [General]
`키워드`: 데이터 캡슐화, 패킷 구조, 헤더, 트레일러, FCS

**Comment by cutehumanS2** [comment]
데이터 캡슐화 과정에서 헤더와 트레일러가 모두 붙는 계층은 **데이터 링크 계층**(OSI 2계층)입니다.

데이터 링크 계층에서 트레일러가 붙는 이유는, 프레임의 무결성을 보장하기 위해 오류 검출을 수행하기 때문입니다.

송신 측에서 데이터를 프레임으로 감싸면서 트레일러에 FCS(Frame Check Sequence) 값을 추가하면, 수신 측에서 프레임을 수신한 후 FCS 값을 검증하여 데이터가 전송 중 손상되었는지 확인합니다.
